src/cla/tcp.rs

```diff
@@ -0,0 +1,511 @@
+                return Ok(true);
+            }
+            TcpClPacket::XferSeg(data) => {
+                if data.flags.contains(XferSegmentFlags::END)
+                    && !data.flags.contains(XferSegmentFlags::START)
+                {
+                    warn!("Received XferSeg with end flag, unexpected in idle mode");
+                }
+                if data.flags.is_empty() {
+                    warn!("Received XferSeg with no flags, unexpected in idle mode");
+                }
+                let mut vec = data.buf.to_vec();
+                self.sender_task
+                    .send(TcpClPacket::XferAck(XferAckData {
+                        tid: data.tid,
+                        len: data.len,
+                        flags: XferSegmentFlags::empty(),
+                    }))
+                    .await?;
+                let mut connection_closing = false;
+                if !data.flags.contains(XferSegmentFlags::END) {
+                    let mut len = data.len;
+                    if data.len > self.data_local.segment_mru {
+                        self.sender_task
+                            .send(TcpClPacket::XferRefuse(XferRefuseData {
+                                reason: XferRefuseReasonCode::NotAcceptable,
+                                tid: data.tid,
+                            }))
+                            .await?;
+                    }
+                    let mut transfer_finished = false;
+                    loop {
+                        if let Some(packet) = self.receiver_task.recv().await {
+                            match packet {
+                                TcpClPacket::SessTerm(mut data) => {
+                                    connection_closing = true;
+                                    data.flags |= SessTermFlags::REPLY;
+                                    self.sender_task.send(TcpClPacket::SessTerm(data)).await?;
+                                }
+                                TcpClPacket::XferSeg(data) => {
+                                    vec.append(&mut data.buf.to_vec());
+                                    len += data.len;
+                                    transfer_finished = data.flags.contains(XferSegmentFlags::END);
+                                    self.sender_task
+                                        .send(TcpClPacket::XferAck(XferAckData {
+                                            tid: data.tid,
+                                            len,
+                                            flags: XferSegmentFlags::empty(),
+                                        }))
+                                        .await?;
+                                }
+                                TcpClPacket::MsgReject(data) => {
+                                    warn!("Received message reject: {:?}", data);
+                                }
+                                _ => {
+                                    error!("Unexpected packet");
```

Comment:
add a debug print giving a few more information, packet bytes, magic numbers, first 10 bytes  or something to debug the issue